packages/blade/src/components/DatePicker/Calendar.web.tsx

```diff
@@ -64,7 +66,18 @@ const Calendar = <Type extends DateSelectionType>({
 
   const dateContext = useDatesContext();
   const isMobile = useIsMobile();
-  const currentDate = _date ?? shiftTimezone('add', new Date());
+  const currentDate = React.useMemo(() => {
```

Comment:
Can you explain whats happening here? the `date` prop as is should work?

Reply:
We are already using dateProp. Currently, _date is passed to Date.
Whenever the user clicks on the next month/year, we update _date, and the date picker focuses on _date.
I modified this logic slightly:
If _date exists, focus on it.
Otherwise, focus on oldValue.
If oldValue is also unavailable, focus on currentDate.

Reply:
what is oldValue?

Reply:
It's the last applied value in the date picker.
Whenever you click/select any value on the date picker, its value changes. But if you click on Cancel, we revert back to the older value using oldValue. When you click on Apply, we save the applied value in oldValue to retrieve it next time.